song_match/song_robot.py

```diff
@@ -48,11 +49,56 @@ def __init__(self, robot: Robot, song: Song):
         await note_cube.blink_and_play_note()
         await self.__wait_until_animation_finished()
 
+    async def play_note_with_error(self, note: Note, round: int):
+        """Make Cozmo play a note.
+
+        :param note: The :class:`~song_match.song.note.Note` to play.
+        :return: None
+        """
+        difficulty = .99
+        rand = random()
+
+        round_difficulty = difficulty ** round
+
+        cube_id = self._song.get_cube_id(note)
+
+        if round_difficulty < rand:
+            cozmoError = True
+            if cube_id == 1:
```

Comment:
We should the use constants for the Cube ID's from Cozmo's SDK here:

* `cozmo.objects.LightCube1Id`
* `cozmo.objects.LightCube2Id`
* `cozmo.objects.LightCube3Id`

http://cozmosdk.anki.com/docs/generated/cozmo.objects.html

Reply:
You could probably remove this entire if else construct by using modulus to increment the `cube_id` and restrict it's range to `1 - 3`.